servers/store/src/test/java/org/projectnessie/server/store/TestStoreWorker.java

```diff
@@ -69,18 +69,18 @@ void testSerde(Map.Entry<ByteString, Content> entry) {
   @Test
   void testSerdeIceberg() {
     String path = "foo/bar";
-    IcebergTable table = IcebergTable.of(path, "xyz", ID);
+    IcebergTable table = IcebergTable.of(path, "x", ID);
 
     ObjectTypes.Content protoTableGlobal =
         ObjectTypes.Content.newBuilder()
             .setId(ID)
-            .setIcebergGlobal(IcebergGlobal.newBuilder().setIdGenerators("xyz"))
+            .setIcebergMetadataPointer(
```

Comment:
I think if we keep this design. All the snapshot id will be present in current metadata pointer of table@ref.
So, this can cause the below problems.
a) Each branch, each table will lose the snapshot history. so time travel within the branch is not possible
b) Each branch, each table will not know the list of all the reachable files as snapshot history is lost. So, if the table is dropped, we cannot know which files to clean for that table@ref.
c) we lose table evolution history per branch, we cannot know how schema, sort order, partition spec is evolved for that table@ref.
d) table properties configured for table@ref will be used for the same table in other reference also. which is not good.

This is high level things I can recall. Yet to analyze how compaction per branch or other Iceberg operations are affected by this.

Reply:
> a) In each branch, each table will lose the snapshot history. So, time travel within the branch is not possible.

Time travel would still be possible via Nessie.

> b) In each branch, each table will not know the list of all the reachable files as snapshot history is lost.

Can you elaborate why `TableMetadata.snapshotLog()` is required for this?

> So, if the table is dropped in that branch, we cannot know which files to clean for that table@ref.

You can still figure that out by traversing the commit-log.

The only situation I can currently imagine is that you create a table on a new branch, add data and then just drop the branch. But every approach would run into this situation.

> c) we lose table evolution history per branch. We cannot know how schema, sort order, partition spec is evolved for that table@ref.

It's still there via the Nessie commits + the information in the referenced snapshots, right?

> d) Table properties configured for table@ref will be visible and used for the same table in other reference also. which can cause unexpected behaviour in other references for the table.

I don't think that there's an issue with exposing other references' information. Proper isolation would require way more than just thinking about it here.

What would those unexpected behaviors be?

> e) we lose metadata-log history per table in a branch. metadata-log is used for deleting old metadata files (and fixed number of maximum metadata files at given point of time). we have an auto action for cleaning the metadata files.

Why would we lose that metadata-log (`TableMetadata.previousFiles()`)?

There will be exactly one current and valid metadata (file).

Reply:
> b) In each branch, each table will not know the list of all the reachable files as snapshot history is lost.
> 
> Can you elaborate why TableMetadata.snapshotLog() is required for this?
I meant Table.snapshots(), which will have all branch's snapshot. So, when I do drop table on branch. At Iceberg side, I cannot know what all snapshots to delete for that branch.

Reply:
> d) Table properties configured for table@ref will be visible and used for the same table in other reference also. which can cause unexpected behaviour in other references for the table.

> I don't think that there's an issue with exposing other references' information. Proper isolation would require way more than just thinking about it here.
> 
> What would those unexpected behaviors be?

For example, If table in branch_1 is configured to use `write.target-file-size-bytes` = 10 MB, same table in all the branches will use that for writing file size. which may be unexpected for same table in other branch as they might be created for other experiments.

Reply:
> e) we lose metadata-log history per table in a branch. metadata-log is used for deleting old metadata files (and fixed number of maximum metadata files at given point of time). we have an auto action for cleaning the metadata files.
> 
> Why would we lose that metadata-log (TableMetadata.previousFiles())?

> There will be exactly one current and valid metadata (file).

I meant we lose per branch, not for overall table. Table in one ref now contains `TableMetadata.previousFiles()` which has metadata files from all the branches. So, when we add new commit in one branch, it might clean up other branch's metadata file when this configuration is enabled.

Reply:
> Now assume branch_1 has 1 commit and branch_2 has 99 commits. If metadata files to retain is configured as 100. For a new commit happens on branch_2, we delete branch_1's metadata file even though it is referenced in Nessie branch_1.

You are correct ajantha. We should be setting this property at table create time and running checks to ensure it isn't changed

> d) Table properties configured for table@ref will be visible and used for the same table in other reference also. which can cause unexpected behaviour in other references for the table

My thinking here is that for the time being table proeprties are global and when we come across a use case which requires table properties per branch we can adjust.

Reply:
on the topic of time travel and history generally: we are in a weird position currently as there are two timelines (iceberg via snapshots and Nessie via commits) and both are accessible via different APIs. We should strive to have 1 history API for a Nessie table ( which uses Nessie's history). So I am not overly worried about the iceberg timeline and iceberg apis to access it, the iceberg timeline may be goofy but the Nessie timeline is the source of truth. We do have to focus on extending Iceberg to ensure that certain API calls are routed to Nessie, these should be generic extension points in the iceberg metadata handling code and should make sense for all extenders of Iceberg (not just Nessie). Make sense?